easybuild/easyblocks/g/gurobi.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+##
+# Copyright 2009-2016 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing Gurobi, implemented as an easyblock
+
+@author: Bob Dröge (University of Groningen)
+"""
+
+import os
+import shutil
+
+from easybuild.easyblocks.generic.tarball import Tarball
+from easybuild.tools.build_log import EasyBuildError
+
+
+class EB_Gurobi(Tarball):
+    """Support for installing linux64 version of Gurobi."""
+
+    def configure_step(self):
+        """No configuration for Gurobi."""
+        # ensure a license file is specified
+        if self.cfg['license_file'] is None:
+            raise EasyBuildError("No license file specified.")
```

Comment:
if `license_file` is defined but points to a non-existing file, you get a nasty traceback like below, so maybe also check here whether the file exists?

```python
if self.cfg['license_file'] is None or not os.path.exists(self.cfg['license_file']):
    raise EasyBuildError("No existing license file specified: %s", self.cfg['license_file'])
```
ERROR: Traceback (most recent call last):
  File "/user/scratchdelcatty/gent/gvo000/gvo00002/vsc40023/easybuild_easy_installed/lib/python2.6/site-packages/easybuild_framework-2.9.0.dev0-py2.6.egg/easybuild/main.py", line 114, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_easy_installed/lib/python2.6/site-packages/easybuild_framework-2.9.0.dev0-py2.6.egg/easybuild/framework/easyblock.py", line 2358, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_easy_installed/lib/python2.6/site-packages/easybuild_framework-2.9.0.dev0-py2.6.egg/easybuild/framework/easyblock.py", line 2274, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_easy_installed/lib/python2.6/site-packages/easybuild_framework-2.9.0.dev0-py2.6.egg/easybuild/framework/easyblock.py", line 2153, in run_step
    step_method(self)()
  File "/tmp/eb-X2qMnA/included-easyblocks/easybuild/easyblocks/gurobi.py", line 56, in install_step
    shutil.copy2(self.cfg['license_file'], lic_path)
  File "/usr/lib64/python2.6/shutil.py", line 95, in copy2
    copyfile(src, dst)
  File "/usr/lib64/python2.6/shutil.py", line 50, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 2] No such file or directory: '/user/home/gent/vsc400/vsc40023/licenses/Gurobi/gurobi.lic'
```